tests/sentry/api/endpoints/test_organization_metrics.py

```diff
@@ -1343,6 +1343,20 @@ def test_crash_free_rate_when_no_session_metrics_data_exist(self):
         assert group["series"]["sum(sentry.sessions.session)"] == [0]
         assert group["series"]["session.crash_free_rate"] == [None]
 
+    @with_feature(FEATURE_FLAG)
+    def test_crash_free_rate_when_no_session_metrics_data_with_orderBy_and_groupBy(self):
```

Comment:
I had trouble understanding how the changes relate to the test so I ran it locally.

this test passes even if the changes in src/ are reverted. Are you sure it tests the thing you're fixing?

(also the rest of the file spells it `orderby` instead of `orderBy`... I wasn't sure at first whether pytest would even pick up the test)

Reply:
This test is unrelated.. I just added it to ensure that the function would not fail under these circumstances

Reply:
ok can you add a test for the first fix then?

Reply:
Actually I was mistaken,  this test does cover this case.. how did you run it and not fail? 
I get when I revert the change
```
  File "/Users/ahmedetefy/Desktop/Dev/sentry/src/sentry/snuba/metrics/datasource.py", line 277, in get_series
    initial_snuba_query = next(iter(snuba_queries.values()))["totals"]
StopIteration

```

Reply:
@untitaker since I now did another commit, running `git checkout HEAD~2 src/ ` will fail the test

Reply:
I don't get it with `pytest tests/sentry/api/endpoints/test_organization_metrics.py`, but do get it when running the test specifically. I believe that's because you have some calls to `DERIVED_METRICS.update` which affect unrelated tests.

Reply:
> I believe that's because you have some calls to DERIVED_METRICS.update which affect unrelated tests.

No, this should not affect any other case actually, and even if the change to `DERIVED_METRICS` state is leaked .. it should not matter unless you are specifically querying `crash_free_fake`

> I don't get it with pytest tests/sentry/api/endpoints/test_organization_metrics.py, but do get it when running the test specifically

Also I ran both all tests and this specific test

Reply:
tldr: in the same PR, ahmed already removed a cache on the global derived metric objects that incorrectly cached the entity type of a derived metric across multiple testruns. so reverting all of `src/` will produce flaky tests, but reverting only the bugfix wrt StopIteration won't.